src/server/routes/sso.js

```diff
@@ -72,13 +72,16 @@ router.get("/google", async function (req, res) {
         if (refererOrigin !== config.server.selfDomain) {
             await dbOps.trackstate(state, refererOrigin, refererUrl);
         }
-    }
 
-    // Redirects to Google's `/authorize` endpoint
-    googleSso.authorize(res, dbOps, googleSso.options, state).then(null, (error) => {
-        console.log(error);
-        res.status(403).json({"isError": true, "message": error.message});
-    });
+        // Redirects to Google's `/authorize` endpoint
+        googleSso.authorize(res, dbOps, googleSso.options, state).then(null, (error) => {
+            console.log(error);
+            res.status(403).json({"isError": true, "message": error.message});
+        });
+    } else {
+        res.status(403).json({"isError": true, "message": "Missing the referrer information"});
```

Comment:
Something more specific like "Authorization request is missing the required 'referer' header"?